0099-group-challenge.md

```diff
@@ -0,0 +1,148 @@
+
+# Motivation
+[motivation]: #motivation
+
+As mentioned in the summary, we'd like to enable lighter gateway nodes without having them need to carry the full (or even a partial) chain.
+Moving PoC challenging onto the consensus group allows for two or more classes of nodes that can all have different roles in the network.
+Having lighter gateways will allow people to participate more cheaply, since the gateways can be cheaper than the current pi-based gateways, which in turn should allow the network to expand and gain value more quickly.
+
+# Stakeholders
+[stakeholders]: #stakeholders
+
+**Who is affected by this HIP?**
+
+Everyone who uses or in the future might want to use the network.
+
+**How are we soliciting feedback on this HIP from these stakeholders?**
+
+This HIP process, initially the community, and reaching out to commercial partners who might prefer to provide lighter and simpler nodes.
+
+# Detailed Explanation
+[detailed-explanation]: #detailed-explanation
+
+## Overview
+
+Currently, all Hotspots are required to validate the full chain block by block in order to participate in the PoC process.
+The run both sides of it, issuing challenges at regular intervals and then gathering up the replies.
+This works well enough currently, but requires some heavy work on the side of the challenging spot, and requires the presence of an up-to-date chain in several places.
+The first iteration of this change would add a new type of gateway that is a member of the p2p mesh but does not have access to the chain except via API calls to external services.
+These nodes would be expected to listen for challenge requests delivered from the consensus group via p2p and then execute them in a timely manner.
+These p2p request paths could include both light and full nodes.
+Challenges would instead come from members of the consensus group, who would divide up the responsibilities for communicating these requests amongst themselves (see the section on [request generation](#request-generation) for more details).
+The members of the group would also keep track of the hops and witnesses much like current miner nodes.
+They would then submit, along with their regular transaction proposals, either a normal receipt or a [voting transaction](#voting-transactions) depending on what design we decide on.
+
+## Challenge Generation
+
+This will likely remain the same as currently, but instead of generating a single challenge, we will generate the entire list of challenges for this particular block, the size of which will be governed by a new algorithm for automatically scaling the number of requests to the size of the network (see [Appendix B](#appendix-b) for details).
+We'll use the same "deterministically random" approach that we use now, just for more iterations.
+Once the list of targets is generated, group members will select a subset of which to construct and deliver requests to.
+
+
+## Request Generation
+[request-generation]: #request-generation
```

Comment:
There's not much about the actual generation of a request here. In the current model the request is generated from the combination of chain entropy and a secret ephemeral key generated by the challenger, which is only revealed in the receipt. How do we plan to replicate this on the consensus group?

Reply:
that's a good question.  I'd forgotten that particular detail when I wrote this.  Maybe the right solution is that everyone in the group races at random?  First to deliver a key and path to a spot wins?

Reply:
I think there's a lot of ways we could do this and we should try to decide on the best way. For example the consensus members could simply propose a set of ephemeral keys per block, and we pick a subset of those (depending on the poc rate), combine them with the block hash  and use that use for pathing. We could also still allow the challenger to submit a key, or we could do some wild secure multiparty compute over elliptic curves (likely a huge amount of work).